cargo-shuttle/src/lib.rs

```diff
@@ -330,54 +331,59 @@ impl Shuttle {
 
-        // 2. Ask for project name
-        if project_args.name.is_none() {
+        // 2. Ask for project name or validate the given one
+        if needs_name {
             printdoc! {"
                 What do you want to name your project?
                 It will be hosted at ${{project_name}}.shuttleapp.rs, so choose something unique!
                 "
             };
-            let client = self.client.as_ref().unwrap();
-            loop {
-                // not using validate_with due to being blocking
-                let p: String = Input::with_theme(&theme)
+        }
+        let mut prev_name: Option<String> = None;
+        loop {
+            // prompt if interactive
+            let name: String = if let Some(name) = project_args.name.clone() {
+                name
+            } else {
+                // not using `validate_with` due to being blocking.
+                Input::with_theme(&theme)
                     .with_prompt("Project name")
-                    .interact()?;
-                match client.check_project_name(&p).await {
-                    Ok(true) => {
-                        println!("{} {}", "Project name already taken:".red(), p);
-                        println!("{}", "Try a different name.".yellow());
-                    }
-                    Ok(false) => {
-                        project_args.name = Some(p);
-                        break;
-                    }
-                    Err(e) => {
-                        // If API error contains message regarding format of error name, print that error and prompt again
-                        if let Ok(api_error) = e.downcast::<ApiError>() {
-                            // If the returned error string changes, this could break
-                            if api_error.message.contains("Invalid project name") {
-                                println!("{}", api_error.message.yellow());
-                                println!("{}", "Try a different name.".yellow());
-                                continue;
-                            }
-                        }
-                        // Else, the API error was about something else.
-                        // Ignore and keep going to not prevent the flow of the init command.
-                        project_args.name = Some(p);
-                        println!(
-                            "{}",
-                            "Failed to check if project name is available.".yellow()
-                        );
-                        break;
-                    }
-                }
+                    .interact()?
+            };
+            // handle forced name on retires
+            if needs_name && prev_name.as_ref().is_some_and(|prev| prev == &name) {
+                project_args.name = Some(prev_name.unwrap());
+                break;
+            }
+            // skip validation if forced
+            if args.force_name {
+                project_args.name = Some(name);
```

Comment:
```suggestion
            let force_name = args.force_name || (needs_name && prev_name.as_ref().is_some_and(|prev| prev == &name));
            if force_name {
                project_args.name = Some(name);
                break;
            }
```

Shorter still :smile: